### PR TITLE
feature/001-Init

### PR DESCRIPTION
**Pull request by fedor-g**
Partly completed hw#1

### CODE REVIEW

File: src/components/Courses/Courses.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+function defineAuthors(id: Array<string>, mockedAuthorsList) {
+	for (let i = 0; i < id.length; i++) {
+		mockedAuthorsList.forEach((au) => {
+			if (id[i] === au.id) {
+				id[i] = au.name;
+			}
+		});
+	}
+	return id;
```
Comment (supercilium) [inline src/components/Courses/Courses.tsx:29]: Looks like you're mutating an input parameter of this function - it is kind a bad practice. Consider using pure functions as much as possible (https://www.freecodecamp.org/news/what-is-a-pure-function-in-javascript-acb887375dfe/)

Could you specify what exactly this function should do? If you need to get names for ids I can suggest you the following
return ids.map(id => mockedAuthorsList.find(author => author.id === id).name)

map function is pure, it does not mutate the input, it creates a new array for the result

you can find more Array built in methods here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods

File: src/components/Courses/Courses.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+function checkCourses(
```
Comment (supercilium) [inline src/components/Courses/Courses.tsx:32]: It is a good approach to pass parameters to a function as an object if you have more than 2 of them

File: src/common/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import styles from './button.module.scss';
+
+export const Button = (props) => {
```
[the remark this answers is not in the document]
Comment (supercilium) [reply src/common/Button/Button.tsx:4]: And also it is better to extend DOM attributes for such low-level components.
image
so you can pass props this way
image
And you will see TS suggestions as well

File: src/components/Courses/Courses.tsx (hunk 1 above)
Comment (fedor-g) [reply src/components/Courses/Courses.tsx:29]: I found the problem in returning a new array - though comparing worked correctly, I received an empty result. The provided solution leads to an empty page =(

Comment (supercilium) [reply src/components/Courses/Courses.tsx]: I tried to use it in your code, looks like it works for the first render
image
but undefined for the following which is triggered by StrictMode (in your index.js file)
btw it should have .ts extention :) 

so either nee to check why re-render break this function or remove strict mode

File: src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { Header } from './components/Header/Header';
+import { Courses } from './components/Courses/Courses';
+import { mockedAuthorsList, mockedCoursesList } from 'src/constants';
+import styles from './app.scss';
+import { CourseInfo } from './components/CourseInfo/CourseInfo';
+
+function App() {
+	const [info, setInfo] = useState(true);
+
+	function toggleInfo(courseId: string) {
```
Comment (supercilium) [inline src/App.tsx:11]: do you need parameter here?

File: src/components/Courses/Courses.tsx (hunk 1 above)
Comment (fedor-g) [reply src/components/Courses/Courses.tsx]: thanks! strict mode removal really helped

File: src/App.tsx (hunk 4 above)
Comment (fedor-g) [reply src/App.tsx:11]: Yep, to know - which info card to show

**Pull request merged**